UpgradeNextMajorParentVersion fails for the `sonargraph-integration` because of the Stapler migration

### Plugin Modernizer version

plugin modernizer 999999-SNAPSHOT (2025-01-28T08:38:19Z)

### What Operating System are you using

Debian / WSL2

### Reproduction steps

`java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar run --plugins sonargraph-integration --recipe UpgradeNextMajorParentVersion`

When I try to compile after the failure the project in `.cache`, I get:

```
[INFO] Some messages have been simplified; recompile with -Xdiags:verbose to get full output
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/AbstractHTMLAction.java:[47,25] no suitable method found for generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse,com.hello2morrow.sonargraph.integration.jenkins.controller.AbstractHTMLAction)
    method org.kohsuke.stapler.HttpResponse.generateResponse(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse,java.lang.Object) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
    method hudson.model.DirectoryBrowserSupport.generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2,java.lang.Object) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineChartAction.java:[107,35] incompatible types: org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineDiffAction.java:[101,25] no suitable method found for generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse,com.hello2morrow.sonargraph.integration.jenkins.controller.PipelineDiffAction)
    method org.kohsuke.stapler.HttpResponse.generateResponse(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse,java.lang.Object) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
    method hudson.model.DirectoryBrowserSupport.generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2,java.lang.Object) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/Plotter.java:[125,18] no suitable method found for doPng(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse)
    method hudson.util.Graph.doPng(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
    method hudson.util.Graph.doPng(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse) is not applicable
      (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
[INFO] 4 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.841 s
[INFO] Finished at: 2025-01-28T10:26:08+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project sonargraph-integration: Compilation failure: Compilation failure:
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/AbstractHTMLAction.java:[47,25] no suitable method found for generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse,com.hello2morrow.sonargraph.integration.jenkins.controller.AbstractHTMLAction)
[ERROR]     method org.kohsuke.stapler.HttpResponse.generateResponse(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse,java.lang.Object) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
[ERROR]     method hudson.model.DirectoryBrowserSupport.generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2,java.lang.Object) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineChartAction.java:[107,35] incompatible types: org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineDiffAction.java:[101,25] no suitable method found for generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse,com.hello2morrow.sonargraph.integration.jenkins.controller.PipelineDiffAction)
[ERROR]     method org.kohsuke.stapler.HttpResponse.generateResponse(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse,java.lang.Object) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
[ERROR]     method hudson.model.DirectoryBrowserSupport.generateResponse(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2,java.lang.Object) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
[ERROR] /home/poddingue/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources/src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/Plotter.java:[125,18] no suitable method found for doPng(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse)
[ERROR]     method hudson.util.Graph.doPng(org.kohsuke.stapler.StaplerRequest2,org.kohsuke.stapler.StaplerResponse2) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerResponse cannot be converted to org.kohsuke.stapler.StaplerResponse2)
[ERROR]     method hudson.util.Graph.doPng(org.kohsuke.stapler.StaplerRequest,org.kohsuke.stapler.StaplerResponse) is not applicable
[ERROR]       (argument mismatch; org.kohsuke.stapler.StaplerRequest2 cannot be converted to org.kohsuke.stapler.StaplerRequest)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
poddingue@PC-bverachten ~/.cache/jenkins-plugin-modernizer-cli/sonargraph-integration/sources (plugin-modernizer/upgradenextmajorparentversion)$ git status
On branch plugin-modernizer/upgradenextmajorparentversion
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   pom.xml
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/AbstractHTMLAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/InvisibleReportAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineChartAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/PipelineDiffAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/Plotter.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/SonargraphDiffAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/SonargraphIntegrationAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/SonargraphReportAction.java
        modified:   src/main/java/com/hello2morrow/sonargraph/integration/jenkins/controller/SonargraphReportBuilder.java
```

### Expected Results

No error.

--- a/plugin-modernizer-core/src/main/resources/META-INF/rewrite/recipes.yml
+++ b/plugin-modernizer-core/src/main/resources/META-INF/rewrite/recipes.yml
@@ -153,6 +153,9 @@ recipeList:
   - org.openrewrite.java.ChangeType:
       oldFullyQualifiedTypeName: org.kohsuke.stapler.StaplerRequest
       newFullyQualifiedTypeName: org.kohsuke.stapler.StaplerRequest2
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.kohsuke.stapler.StaplerResponse
+      newFullyQualifiedTypeName: org.kohsuke.stapler.StaplerResponse2
   - org.openrewrite.java.ChangeMethodName:
       methodPattern: org.kohsuke.stapler.Stapler getCurrentRequest()
       newMethodName: getCurrentRequest2

--- a/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/recipes/DeclarativeRecipesTest.java
+++ b/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/recipes/DeclarativeRecipesTest.java
@@ -1420,10 +1420,12 @@ public class DeclarativeRecipesTest implements RewriteTest {
                                 import javax.servlet.ServletException;
                                 import org.kohsuke.stapler.Stapler;
                                 import org.kohsuke.stapler.StaplerRequest;
+                                import org.kohsuke.stapler.StaplerResponse;
 
                                 public class Foo {
                                     public void foo() {
                                         StaplerRequest req = Stapler.getCurrentRequest();
+                                        StaplerResponse response = Stapler.getCurrentResponse();
                                     }
                                 }
                                 """,
@@ -1431,10 +1433,12 @@ public class DeclarativeRecipesTest implements RewriteTest {
                                 import jakarta.servlet.ServletException;
                                 import org.kohsuke.stapler.Stapler;
                                 import org.kohsuke.stapler.StaplerRequest2;
+                                import org.kohsuke.stapler.StaplerResponse2;
 
                                 public class Foo {
                                     public void foo() {
                                         StaplerRequest2 req = Stapler.getCurrentRequest2();
+                                        StaplerResponse2 response = Stapler.getCurrentResponse2();
                                     }
                                 }
                                 """)));

